Couchpotato: init at 3.0.1 + couchpotato module

fadenb opened the pull request:
###### Motivation for this change


###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -1178,6 +1178,45 @@ in {
     };
   };
 
+  couchpotato = buildPythonPackage rec {
```
couchpotato is used as an application and not as a library. Therefore, it doesn't belong in `python-packages.nix`. Instead, it should fit somewhere else in Nixpkgs tree based on what it is/does.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -1178,6 +1178,45 @@ in {
+      sha256 = "1xwjis3ijh1rff32mpdsphmsclf0lkpd3phpgxkccrigq1m9r3zh";
+    };
+
+    phases = [ "unpackPhase" "patchPhase" "installPhase" "postInstall" ];
```
please remove this since there is no need for it

FRidh replied:
you might want to pass `format = "other";`. That gives you a `buildPythonPackage` for non-setuptools installations.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -1178,6 +1178,45 @@ in {
+      substituteInPlace CouchPotato.py --replace "dirname(os.path.abspath(__file__))" "os.path.join(dirname(os.path.abspath(__file__)), '../${python.sitePackages}')"
+    '';
+
+    installPhase = ''
```
you're making a proper package out of it. That's good. I wish upstream would do that as well.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -1178,6 +1178,45 @@ in {
+      cp -r libs/* $out/${python.sitePackages}/
+      cp -r couchpotato $out/${python.sitePackages}/
+
+      cp CouchPotato.py $out/bin/
```
having a binary with an extension is rather ugly. Maybe we want to rename it instead to `couchpotato` ?

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -1178,6 +1178,45 @@ in {
+    '';
+
+    postInstall = ''
+      wrapProgram "$out/bin/CouchPotato.py" --prefix PYTHONPATH : "$PYTHONPATH:$out/${python.sitePackages}" \
```
the wrapping is done in the `fixupPhase` so this shouldn't be necessary.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -1178,6 +1178,45 @@ in {
+      description = "Automatic movie downloading via NZBs and torrents";
+      license     = licenses.gpl3;
+      maintainers = with maintainers; [ fadenb ];
+    };
```
homepage?

fadenb commented:
@FRidh: Thanks for your input. I tried to address all the issues:
* homepage: Added to metadata
* .py extension: Removed and lowercased to 'couchpotato'
* phases: Did not know about format :)
* not a library: Moved to the normal package tree
* fixup/wrapping: The automatic fixup wraps way more than just the CouchPotato.py resulting in errors as the code tries to load the shell wrapper as python. Additionally the automatic wrapper does not add ```$out/${python.sitePackages}``` to the `$PYTHONPATH` so I kept my original wrapper call but moved it to the fixupPhase.

FRidh commented:
Great. But then at least use `set` instead of `prefix`.

FRidh commented on pkgs/servers/couchpotato/default.nix:
```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchurl, pythonPackages, pkgs, lib, ... }:
```
don't pass in `pkgs`

FRidh replied:
stdenv isn't needed either, nor are the dots.

FRidh commented:
> phases: Did not know about format :)

That's still undocumented actually.

fadenb committed:
couchpotato: init at 3.0.1

fadenb committed:
couchpotato module: init
